### DISCUSSION TITLE
Question on alias in config

### DISCUSSION THREAD

**Discussion by govardha** [kannibalox/pyrosimple#68] [General]
Thank you @kannibalox for this much needed port into python3.  I finally am getting the flow fully and have successfully done my 2nd seedbox migration.  I can't thank you enough for the time & effort you have put into this project.  I along with a handful of other nerds I am sure greatly appreciate it.

I did notice an issue, in the pyrossimple config file I set an alias for a given tracker.  But when I use rtcontrol with that tracker alias I get 0 results.  If I run the same query with actual tracker url rtcontrol gives me all the results perfectly.  What could I be doing wrong?

**Comment by kannibalox** [comment]
Can you post an example of your config and the commands you're running? E.G. if you have the following `config.toml`:
```toml
[ALIASES]
EX = ["example.com"]
```
then `rtcontrol alias=EX` and `rtcontrol tracker=/example.com/` should return the same values like you're expecting. You can use `rtcontrol tracker=/example.com/ -o alias` to see what the stored alias value is.

The alias value is cached, so if it's not what you're expecting (perhaps from testing or a previous config), you can reset it with:
```
rtcontrol --custom memo_alias= // -o hash
rtcontrol // -o alias
```

**Reply by govardha** [reply]
Here you go.

```
rtcontrol tracker=/xyz/ -qo alias

MOUSE   https://xyz.org/tracker.php/xxxxx/announce
MOUSE   https://xyz.org/tracker.php/xxxxx/announce
xyz.org https://xyz.org/tracker.php/xxxxx/announce
xyz.org https://xyz.org/tracker.php/xxxxx/announce
MOUSE   https://xyz.org/tracker.php/xxxxx/announce
MOUSE   https://xyz.org/tracker.php/xxxxx/announce
MOUSE   https://xyz.org/tracker.php/xxxxx/announce
MOUSE   https://xyz.org/tracker.php/xxxxx/announce
MOUSE   https://xyz.org/tracker.php/xxxxx/announce
MOUSE   https://xyz.org/tracker.php/xxxxx/announce
MOUSE   https://xyz.org/tracker.php/xxxxx/announce
MOUSE   https://xyz.org/tracker.php/xxxxx/announce
MOUSE   https://xyz.org/tracker.php/xxxxx/announce
xyz.org https://xyz.org/tracker.php/xxxxx/announce
MOUSE   https://xyz.org/tracker.php/xxxxx/announce
xyz.org https://xyz.org/tracker.php/xxxxx/announce
xyz.org https://xyz.org/tracker.php/xxxxx/announce
MOUSE   https://xyz.org/tracker.php/xxxxx/announce
```

Snippet from config.toml

```
[ALIASES]
MOUSE = ["https://xyz.org/tracker.php/xxxxx/announce"]
```

**Reply by kannibalox** [reply]
`MOUSE = [" https://xyz.org/tracker.php/xxxxx/announce"]`
It looks like the leading space in here is throwing things off, I'm adding a code fix for it but in the meantime if you remove it and reset the aliases as per above, that should fix things.

**Reply by govardha** [reply]
> `MOUSE = [" https://xyz.org/tracker.php/xxxxx/announce"]` It looks like the leading space in here is throwing things off, I'm adding a code fix for it but in the meantime if you remove it and reset the aliases as per above, that should fix things.

I am sorry, when I was copy/pastaing and anonymysing my content, I included the leading empty space, the config actually doesn't have an extra empty space.  Updated my 2nd post to reflect no empty space.

**Reply by kannibalox** [reply]
Hm, in that case I'm a little confused, that same configuration works as expected for me. Can you try resetting the aliases as per above to see if that has any effect? Is that the exact output of `rtcontrol tracker=/xyz/ -qo alias`? I ask because it seems to be including `tracker` as well. Can you post the command you're using to filter by the alias?

**Reply by govardha** [reply]
Ok when I went and looked into the contents of $hash.torrent.rtorrent file and I see the entry memo_alias5:MOUSE11 vs memo_alias5:xyz.org11.  I am not sure how that field got populated into the alias field.

I will try to chtor and change the content of the alias field, unless you have a better idea to reset the alias in the $hash.torrent.rtorrent file.

**Reply by kannibalox** [reply]
This will reset and repopulate the alias without having to use chtor (which, just FYI, would have no effect on a running instance, you'd have to shut rTorrent down, use chtor, and then start it back up):
```
rtcontrol --custom memo_alias= // -o hash
rtcontrol // -o alias
```

if you wanted to just limit it to items with the incorrect alias:
```
rtcontrol --custom memo_alias= alias=xyz.org -o hash
rtcontrol tracker=/xyz/ -o alias
```

You can also add this to config to hopefully prevent it from happening again:
MOUSE = ["xyz.org", "https://xyz.org/tracker.php/xxxxx/announce"]

**Comment by govardha** [comment]
w00t and that fixed it and everything is happy in my alias world & rtcontrol!
